### PR TITLE
darwin: deprecate phases

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#28910

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/stdenv/darwin/default.nix
Comment (toonn) [inline pkgs/stdenv/darwin/default.nix:258]: If we're gonna update the bootstrap-tools archive we should probably do this.

**Review by toonn — COMMENTED** [review]
LGTM for the most part as long as the semantics are retained.

File: pkgs/stdenv/darwin/default.nix
Comment (Stunkymonkey) [reply pkgs/stdenv/darwin/default.nix]: done.

File: pkgs/stdenv/darwin/default.nix
Comment (veprbl) [inline pkgs/stdenv/darwin/default.nix:101]:
```suggestion
      inherit (last.pkgs) runCommandLocal;
```

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 161270` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hello</li>
  </ul>
</details>

**Comment by toonn** [comment]
Wanted to note that since a new bootstrap-tools tarball and a full stdenv rebuild is quite a big pill to swallow for the x86_64-darwin Hydra and this is a fairly low priority change, it would be great if maintainers could find some other stdenv rebuild to pair this with when merging to staging.

(Just a suggestion.)

**Comment by Stunkymonkey** [comment]
I am fine with "additional changes" / wait. I just thought, that using `runCommandLocal` would speedup the build.

**Comment by toonn** [comment]
@Stunkymonkey, any speedups are definitely welcome. It's just that the x86_64-darwin infra always has a hard time catching up with large rebuilds, and this is basically as fundamental as it gets for Darwin but a small change overall. So it would be nice if other changes could be grouped. I have something (#162255) in the works that would cause  a stdenv rebuild too, for example.

**Comment by Artturin** [comment]
rebuild-darwin-stdenv PRs have bene merged lately so it would be a good time to merge these. https://github.com/NixOS/nixpkgs/issues?q=label%3A10.rebuild-darwin-stdenv+sort%3Aupdated-desc+is%3Aclosed

**Comment by Stunkymonkey** [comment]
@ofborg build stdenv

**Comment by uri-canva** [comment]
We've got another darwin stdenv rebuild PR in staging now: https://github.com/NixOS/nixpkgs/pull/170654

@toonn can you approve this so we can merge it? You commented "LGTM for the most part" above so I figure you were almost ready but not quite to approve at the time?

**Comment by uri-canva** [comment]
Also we could come up with a new label to add to PRs that are ready to merge into staging? Then before merging staging into staging next we could merge them all. Otherwise we risk leaving PRs unmerged because we never look at them at the exact times they are ready to merge.

**Review by toonn — APPROVED** [review]
@uri-canva, I agree a way to batch up Darwin stdenv rebuilding changes would be nice. I did not intend to have this pending on my explicit approval.

**Pull request merged**
